### ISSUE TITLE
Bug report

### ISSUE THREAD

**Issue by Mi-Nigle** [PowerShell/PSReadLine#1557]
EDIT: I've just seen the pinned 'ArgumentOutOfRangeException' issue. It looks similar to mine so I'll try the fixes there. Sorry for duplicating issues.
_______________________________________
New to GitHub so apologies if I've not included all information.

I'm also new to coding Python and only used visual basic previously.

Just been using visual studio and an error occurred and it told me to report it with this info. This happens every time I run my code:

```
Last 200 Keys:
 / C h 0 4 / 0 4 _ 0 4 / b e g i n / a p p . p y " Enter
 & Space C : / U s e r s / N i g e l / A p p D a t a / L o c a l / P r o g r a m s / P y t h o n / P y t h o n 3 8 - 3 2 / p y t h o n . e x e Space " d : / U s e r s / N i g e l / P r o g r a m m i n g Space T r a i n i n g / E x _ F i l e s _ P r o g _ F o u n d a t i o n s / E x e r c i s e Space F i l e s / C h 0 4 / 0 4 _ 0 4 / b e g i n / a p p . p y " Enter
 &

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: left
Actual value was -1.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.Internal.VirtualTerminal.set_CursorLeft(Int32 value) 
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)
   at Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
```

It was working fine. I ran my code, altered it slightly, saved it, then ran it again and this happened.

I killed the terminal and tried running my code again. It doesn't run and I get a new error:

```
Oops, something went wrong.  Please report this bug with the details below.     
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 9 Keys:
 & Space C : / U s e r

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: left
Actual value was -1.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.Internal.VirtualTerminal.set_CursorLeft(Int32 value) 
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)
   at Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
PS D:\Users\Nigel\Programming Training\Ex_Files_Prog_Foundations\Exercise Files> s/Nigel/AppData/Local/Programs/Python/Python38-32/python.exe "d:/Users/Nigel/Programming Training/Ex_Files_Prog_Foundations/Exercise Files/Ch04/04_04/begin/app.py"
s/Nigel/AppData/Local/Programs/Python/Python38-32/python.exe : The term 
's/Nigel/AppData/Local/Programs/Python/Python38-32/python.exe' is not 
recognized as the name of a cmdlet, function, script file, or operable 
program. Check the spelling of the name, or if a path was included, verify      
that the path is correct and try again.
At line:1 char:1
+ s/Nigel/AppData/Local/Programs/Python/Python38-32/python.exe "d:/User ...     
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (s/Nigel/AppData...8-32/python.e  
   xe:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS D:\Users\Nigel\Programming Training\Ex_Files_Prog_Foundations\Exercise Files>  C:/User
```

**Comment by Mi-Nigle** [comment]
Sorry for the double post but I killed the terminal and tried running my code again. It doesn't run and I get a new error:

```
Oops, something went wrong.  Please report this bug with the details below.     
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 9 Keys:
 & Space C : / U s e r

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: left
Actual value was -1.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.Internal.VirtualTerminal.set_CursorLeft(Int32 value) 
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)
   at Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
PS D:\Users\Nigel\Programming Training\Ex_Files_Prog_Foundations\Exercise Files> s/Nigel/AppData/Local/Programs/Python/Python38-32/python.exe "d:/Users/Nigel/Programming Training/Ex_Files_Prog_Foundations/Exercise Files/Ch04/04_04/begin/app.py"
s/Nigel/AppData/Local/Programs/Python/Python38-32/python.exe : The term 
's/Nigel/AppData/Local/Programs/Python/Python38-32/python.exe' is not 
recognized as the name of a cmdlet, function, script file, or operable 
program. Check the spelling of the name, or if a path was included, verify      
that the path is correct and try again.
At line:1 char:1
+ s/Nigel/AppData/Local/Programs/Python/Python38-32/python.exe "d:/User ...     
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (s/Nigel/AppData...8-32/python.e  
   xe:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS D:\Users\Nigel\Programming Training\Ex_Files_Prog_Foundations\Exercise Files>  C:/User
```

**Comment by Mi-Nigle** [comment]
Just a small update, I haven't been able to fix this however I've found that if VSC window fills the screen there are no errors. If I reduce the size of the window by say, half, I get the error. 

I was trying to work with a notes window and VSC window side by side but that clearly won't work for some reason.

**Comment by daxian-dbw** [comment]
> Oops, something went wrong.  Please report this bug with the details below.     
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new

This URL indicates you were using a very old version of PSReadLine. This issue was fixed (see #1306). Please upgrade to the 2.0.1 version of PSReadLine from PowerShellGallery. See the [upgrading section](https://github.com/PowerShell/PSReadLine#upgrading) for instructions.

> I'm also new to coding Python and only used visual basic previously.

Please be noted that you were running commands in PowerShell in VS Code. It's likely the Windows PowerShell on your machine, which VS Code uses as the default terminal.
